crates/bevy_ecs/src/storage/sorted_small_vec.rs

```diff
@@ -0,0 +1,259 @@
+use smallvec::SmallVec;
+use core::cmp::Ordering;
+
+/// Stores a sorted list of indices with quick implementation for union, difference, intersection.
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct SortedSmallVec<const N: usize>(SmallVec<[usize; N]>);
+
+impl<const N: usize> IntoIterator for SortedSmallVec<N> {
+    type Item = usize;
+    type IntoIter = <SmallVec<[usize; N]> as IntoIterator>::IntoIter;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.0.into_iter()
+    }
+}
+
+impl<const N: usize> Default for SortedSmallVec<N> {
+    fn default() -> Self {
+        Self::new_const()
+    }
+}
+
+impl<const N: usize> SortedSmallVec<N> {
+    /// Construct an empty vector
+    pub fn new() -> Self {
+        Self(SmallVec::new())
+    }
+
+    /// Construct an empty vector
+    ///
+    /// This is a `const` version of [`SortedSmallVec::new()`]
+    pub(crate) const fn new_const() -> Self {
+        Self(SmallVec::new_const())
+    }
+
+    /// Construct a new `SortedSmallVec` from a `Vec<usize>`.
+    ///
+    /// Elements are copied and put in a sorted order if the original `Vec` isn't ordered.
+    /// Duplicates are removed.
+    pub fn from_vec(vec: Vec<usize>) -> Self {
+        let mut sorted_vec = Self(SmallVec::with_capacity(vec.len()));
+        for value in vec {
+            sorted_vec.insert(value);
+        }
+        sorted_vec
+    }
+
+    pub(crate) fn ones(&self) -> impl Iterator<Item = usize> + '_ {
```

Comment:
`ones` was kinda confusing here. It makes sense in the context of "replacing FixedBitSet", but it doesn't really make sense for this struct.

Can we just say `iter`? That's pretty much all this is.

Reply:
I've elected for a simple rename to iter(), implementing `IntoIterator` for `&SortedVecSet` would have me change `set.ones()` into `(&set).into_iter())` everywhere.

Reply:
> implementing `IntoIterator` for `&SortedVecSet` would have me change `set.ones()` into `(&set).into_iter())` everywhere.

Oh, yes, definitely have an `iter()` method!  I just meant it might make sense to have *both*, like `Vec` does.